src/parsing.jl

```diff
@@ -263,28 +263,28 @@ function parse_numeric_array(eltype::Type{T}, str::AbstractString) where T
             return parse(Int, lower):parse(Int, upper)
         end
 
-        arr = OffsetArray(T, ranges...)
+        arr = OffsetArray{T}(ranges...)
     else
-        arr = Array{T}(array_size(str)...)
+        arr = Array{T}(undef, array_size(str)...)
     end
 
     idx_iter = imap(reverse, product(reverse(indices(arr))...))
     el_iter = eachmatch(r"[^\}\{,]+", str[eq_ind+1:end])
     for (idx, num_match) in zip(idx_iter, el_iter)
-        arr[idx...] = parse(T, num_match.match)
+        arr[idx...] .= parse(T, num_match.match)
     end
 
     return arr
 end
 
 function array_size(str)
-    ndims = findfirst(c -> c != '{', str) - 1
+    ndims = something(findfirst(c -> c != '{', str), 0) - 1
```

Comment:
Maybe remove the `something` and throw an error on `nothing` (like "Invalid PostgreSQL array literal '$str'")

Reply:
Should we also through an error when this is < 0 on 0.6 then?

Reply:
Nah keep it as is, I understand better after https://github.com/invenia/LibPQ.jl/pull/40#discussion_r198328152